src/components/Info/index.js

```diff
@@ -1,21 +1,32 @@
 import React, { useState } from 'react'
 import { format, parse, isPast } from 'date-fns'
-
 import RSVP from './Form'
-
 import { Info, RsvpButton, Blinker, Bouncer } from './elements'
+import { useReducer } from 'react'
+
+const reducer = (state, action) => {
+  const { type, payload } = action
+  return { ...state, [type]: payload }
+}
+
+export const StatusDispatch = React.createContext(null)
 
 export default ({ site, city, info, attendeesNumber }) => {
-  const [open, setOpen] = useState(false)
-  const [submitted, setSubmitted] = useState(false)
-  const [error, setError] = useState({ message: '' })
+
+  const initialState = {
+    open: false,
+    submitted: false,
+    error: { message: '' }
+  }
+
+  const [state, dispatch] = useReducer(reducer, initialState)
```

Comment:
Can you destructure state like 

`const {open, submitted} = state` so that the references to `state.open`, `state.submitted` etc can be abbreviated to just `open`, `submitted`, etc.